internal/provider/data_source_project.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-resource-manager/preview/2019-12-10/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+var projectDefaultTimeout = time.Minute * 1
+
+func dataSourceProject() *schema.Resource {
+	return &schema.Resource{
+		Description: "The project data source provides information about an existing HashiCorp project.",
+		ReadContext: dataSourceProjectRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &projectDefaultTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"name": {
+				Description: "The name of the HCP project.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			// Optional inputs
+			"id": {
```

Comment:
This should be `project_id`, since Terraform reserves `id` for its own unique identifier. I would expect this to be a required attribute, since you can only fetch a project by ID right?

Reply:
That's right